checker/src/main/java/org/checkerframework/checker/determinism/DeterminismAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,990 @@
+import org.checkerframework.checker.determinism.qual.Det;
+import org.checkerframework.checker.determinism.qual.NonDet;
+import org.checkerframework.checker.determinism.qual.OrderNonDet;
+import org.checkerframework.checker.determinism.qual.PolyDet;
+import org.checkerframework.common.basetype.BaseAnnotatedTypeFactory;
+import org.checkerframework.common.basetype.BaseTypeChecker;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.framework.flow.CFAbstractAnalysis;
+import org.checkerframework.framework.flow.CFAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFTransfer;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.source.Result;
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedArrayType;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedDeclaredType;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedExecutableType;
+import org.checkerframework.framework.type.QualifierHierarchy;
+import org.checkerframework.framework.type.poly.QualifierPolymorphism;
+import org.checkerframework.framework.type.treeannotator.ListTreeAnnotator;
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+import org.checkerframework.framework.type.typeannotator.ListTypeAnnotator;
+import org.checkerframework.framework.type.typeannotator.TypeAnnotator;
+import org.checkerframework.framework.util.GraphQualifierHierarchy;
+import org.checkerframework.framework.util.MultiGraphQualifierHierarchy;
+import org.checkerframework.javacutil.AnnotationBuilder;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.TreeUtils;
+import org.checkerframework.javacutil.TypesUtils;
+
+/** The annotated type factory for the determinism type-system. */
+public class DeterminismAnnotatedTypeFactory extends BaseAnnotatedTypeFactory {
+    /** The @NonDet annotation. */
+    public final AnnotationMirror NONDET = AnnotationBuilder.fromClass(elements, NonDet.class);
+    /** The @OrderNonDet annotation. */
+    public final AnnotationMirror ORDERNONDET =
+            AnnotationBuilder.fromClass(elements, OrderNonDet.class);
+    /** The @Det annotation. */
+    public final AnnotationMirror DET = AnnotationBuilder.fromClass(elements, Det.class);
+    /** The @PolyDet annotation. */
+    public final AnnotationMirror POLYDET;
+    /** The @PolyDet("up") annotation. */
+    public final AnnotationMirror POLYDET_UP;
+    /** The @PolyDet("down") annotation. */
+    public final AnnotationMirror POLYDET_DOWN;
+    /** The @PolyDet("use") annotation. */
+    public final AnnotationMirror POLYDET_USE;
+
+    /** The java.util.Set interface. */
+    private final TypeMirror SetInterfaceTypeMirror =
```

Comment:
In Java, only class names should use start with an upper case letter followed by both upper and lower case letter.  So these fields should either be all upper case or start with a lower case letter.